{"text": "2024-10-19 06:30:20 | INFO    | __main__:stash:131 | run - {'image': '20231121', 'version': '0.10.1.post2', 'role': 'EpicPlayer', 'headless': True}\n", "record": {"elapsed": {"repr": "0:00:01.143307", "seconds": 1.143307}, "exception": null, "extra": {"image": "20231121", "version": "0.10.1.post2", "role": "EpicPlayer", "headless": true}, "file": {"name": "claim.py", "path": "/home/runner/work/Epic-Free-Games/Epic-Free-Games/epic/src/claim.py"}, "function": "stash", "level": {"icon": "‚ÑπÔ∏è", "name": "INFO", "no": 20}, "line": 131, "message": "run", "module": "claim", "name": "__main__", "process": {"id": 3641, "name": "MainProcess"}, "thread": {"id": 140225620368256, "name": "MainThread"}, "time": {"repr": "2024-10-19 06:30:20.051560+08:00", "timestamp": 1729290620.05156}}}
{"text": "2024-10-19 06:30:24 | WARNING    | epic_games.agent:get_order_history:467 | Failed to get order history, cookie may have expired - {}\n", "record": {"elapsed": {"repr": "0:00:06.056965", "seconds": 6.056965}, "exception": null, "extra": {}, "file": {"name": "agent.py", "path": "/home/runner/work/Epic-Free-Games/Epic-Free-Games/epic/src/epic_games/agent.py"}, "function": "get_order_history", "level": {"icon": "‚ö†Ô∏è", "name": "WARNING", "no": 30}, "line": 467, "message": "Failed to get order history, cookie may have expired", "module": "agent", "name": "epic_games.agent", "process": {"id": 3641, "name": "MainProcess"}, "thread": {"id": 140225620368256, "name": "MainThread"}, "time": {"repr": "2024-10-19 06:30:24.965218+08:00", "timestamp": 1729290624.965218}}}
{"text": "2024-10-19 06:30:25 | DEBUG    | __main__:create_tasks:66 | Put task - {'title': 'Kardboard Kings', 'url': 'https://store.epicgames.com/en-US/p/kardboard-kings-0962f8'}\n", "record": {"elapsed": {"repr": "0:00:06.251846", "seconds": 6.251846}, "exception": null, "extra": {"title": "Kardboard Kings", "url": "https://store.epicgames.com/en-US/p/kardboard-kings-0962f8"}, "file": {"name": "claim.py", "path": "/home/runner/work/Epic-Free-Games/Epic-Free-Games/epic/src/claim.py"}, "function": "create_tasks", "level": {"icon": "üêû", "name": "DEBUG", "no": 10}, "line": 66, "message": "Put task", "module": "claim", "name": "__main__", "process": {"id": 3641, "name": "MainProcess"}, "thread": {"id": 140225620368256, "name": "MainThread"}, "time": {"repr": "2024-10-19 06:30:25.160099+08:00", "timestamp": 1729290625.160099}}}
{"text": "2024-10-19 06:30:25 | DEBUG    | __main__:create_tasks:66 | Put task - {'title': 'Invincible Presents: Atom Eve', 'url': 'https://store.epicgames.com/en-US/p/atom-eve-ff090d'}\n", "record": {"elapsed": {"repr": "0:00:06.252380", "seconds": 6.25238}, "exception": null, "extra": {"title": "Invincible Presents: Atom Eve", "url": "https://store.epicgames.com/en-US/p/atom-eve-ff090d"}, "file": {"name": "claim.py", "path": "/home/runner/work/Epic-Free-Games/Epic-Free-Games/epic/src/claim.py"}, "function": "create_tasks", "level": {"icon": "üêû", "name": "DEBUG", "no": 10}, "line": 66, "message": "Put task", "module": "claim", "name": "__main__", "process": {"id": 3641, "name": "MainProcess"}, "thread": {"id": 140225620368256, "name": "MainThread"}, "time": {"repr": "2024-10-19 06:30:25.160633+08:00", "timestamp": 1729290625.160633}}}
{"text": "2024-10-19 06:30:29 | INFO    | epic_games.agent:claim_weekly_games:294 | claim_weekly_games - {'action': 'go to store', 'url': 'https://store.epicgames.com/en-US/p/kardboard-kings-0962f8'}\n", "record": {"elapsed": {"repr": "0:00:10.562766", "seconds": 10.562766}, "exception": null, "extra": {"action": "go to store", "url": "https://store.epicgames.com/en-US/p/kardboard-kings-0962f8"}, "file": {"name": "agent.py", "path": "/home/runner/work/Epic-Free-Games/Epic-Free-Games/epic/src/epic_games/agent.py"}, "function": "claim_weekly_games", "level": {"icon": "‚ÑπÔ∏è", "name": "INFO", "no": 20}, "line": 294, "message": "claim_weekly_games", "module": "agent", "name": "epic_games.agent", "process": {"id": 3641, "name": "MainProcess"}, "thread": {"id": 140225620368256, "name": "MainThread"}, "time": {"repr": "2024-10-19 06:30:29.471019+08:00", "timestamp": 1729290629.471019}}}
{"text": "2024-10-19 06:30:42 | INFO    | epic_games.agent:claim_weekly_games:294 | claim_weekly_games - {'action': 'go to store', 'url': 'https://store.epicgames.com/en-US/p/atom-eve-ff090d'}\n", "record": {"elapsed": {"repr": "0:00:23.825305", "seconds": 23.825305}, "exception": null, "extra": {"action": "go to store", "url": "https://store.epicgames.com/en-US/p/atom-eve-ff090d"}, "file": {"name": "agent.py", "path": "/home/runner/work/Epic-Free-Games/Epic-Free-Games/epic/src/epic_games/agent.py"}, "function": "claim_weekly_games", "level": {"icon": "‚ÑπÔ∏è", "name": "INFO", "no": 20}, "line": 294, "message": "claim_weekly_games", "module": "agent", "name": "epic_games.agent", "process": {"id": 3641, "name": "MainProcess"}, "thread": {"id": 140225620368256, "name": "MainThread"}, "time": {"repr": "2024-10-19 06:30:42.733558+08:00", "timestamp": 1729290642.733558}}}
{"text": "2024-10-19 06:30:54 | ERROR    | __main__:run:158 | An error has been caught in function 'run', process 'MainProcess' (3641), thread 'MainThread' (140225620368256): - {}\nTraceback (most recent call last):\n  File \"/home/runner/work/Epic-Free-Games/Epic-Free-Games/epic/src/claim.py\", line 162, in <module>\n    asyncio.run(run())\n  File \"/opt/hostedtoolcache/Python/3.10.15/x64/lib/python3.10/asyncio/runners.py\", line 44, in run\n    return loop.run_until_complete(main)\n  File \"/opt/hostedtoolcache/Python/3.10.15/x64/lib/python3.10/asyncio/base_events.py\", line 636, in run_until_complete\n    self.run_forever()\n  File \"/opt/hostedtoolcache/Python/3.10.15/x64/lib/python3.10/asyncio/base_events.py\", line 603, in run_forever\n    self._run_once()\n  File \"/opt/hostedtoolcache/Python/3.10.15/x64/lib/python3.10/asyncio/base_events.py\", line 1909, in _run_once\n    handle._run()\n  File \"/opt/hostedtoolcache/Python/3.10.15/x64/lib/python3.10/asyncio/events.py\", line 80, in _run\n    self._context.run(self._callback, *self._args)\n> File \"/home/runner/work/Epic-Free-Games/Epic-Free-Games/epic/src/claim.py\", line 158, in run\n    await agent.stash()\n  File \"/home/runner/work/Epic-Free-Games/Epic-Free-Games/epic/src/claim.py\", line 151, in stash\n    await self.claim_epic_games(context)\n  File \"/home/runner/work/Epic-Free-Games/Epic-Free-Games/epic/src/claim.py\", line 122, in claim_epic_games\n    await epic.claim_weekly_games(page, single_promotions)\n  File \"/opt/hostedtoolcache/Python/3.10.15/x64/lib/python3.10/site-packages/tenacity/asyncio/__init__.py\", line 189, in async_wrapped\n    return await copy(fn, *args, **kwargs)\n  File \"/opt/hostedtoolcache/Python/3.10.15/x64/lib/python3.10/site-packages/tenacity/asyncio/__init__.py\", line 111, in __call__\n    do = await self.iter(retry_state=retry_state)\n  File \"/opt/hostedtoolcache/Python/3.10.15/x64/lib/python3.10/site-packages/tenacity/asyncio/__init__.py\", line 153, in iter\n    result = await action(retry_state)\n  File \"/opt/hostedtoolcache/Python/3.10.15/x64/lib/python3.10/site-packages/tenacity/_utils.py\", line 99, in inner\n    return call(*args, **kwargs)\n  File \"/opt/hostedtoolcache/Python/3.10.15/x64/lib/python3.10/site-packages/tenacity/__init__.py\", line 398, in <lambda>\n    self._add_action_func(lambda rs: rs.outcome.result())\n  File \"/opt/hostedtoolcache/Python/3.10.15/x64/lib/python3.10/concurrent/futures/_base.py\", line 451, in result\n    return self.__get_result()\n  File \"/opt/hostedtoolcache/Python/3.10.15/x64/lib/python3.10/concurrent/futures/_base.py\", line 403, in __get_result\n    raise self._exception\n  File \"/opt/hostedtoolcache/Python/3.10.15/x64/lib/python3.10/site-packages/tenacity/asyncio/__init__.py\", line 114, in __call__\n    result = await fn(*args, **kwargs)\n  File \"/home/runner/work/Epic-Free-Games/Epic-Free-Games/epic/src/epic_games/agent.py\", line 310, in claim_weekly_games\n    await expect(cta_btn).to_have_text(\"View In Cart\")\n  File \"/opt/hostedtoolcache/Python/3.10.15/x64/lib/python3.10/site-packages/playwright/async_api/_generated.py\", line 19164, in to_have_text\n    await self._impl_obj.to_have_text(\n  File \"/opt/hostedtoolcache/Python/3.10.15/x64/lib/python3.10/site-packages/playwright/_impl/_assertions.py\", line 481, in to_have_text\n    await self._expect_impl(\n  File \"/opt/hostedtoolcache/Python/3.10.15/x64/lib/python3.10/site-packages/playwright/_impl/_assertions.py\", line 74, in _expect_impl\n    raise AssertionError(\nAssertionError: Locator expected to have text 'View In Cart'\nActual value: <element(s) not found> \nCall log:\nLocatorAssertions.to_have_text with timeout 5000ms\n  - waiting for locator(\"//aside//button[@data-testid='add-to-cart-cta-button']\")\n\n", "record": {"elapsed": {"repr": "0:00:35.138365", "seconds": 35.138365}, "exception": {"type": "AssertionError", "value": "Locator expected to have text 'View In Cart'\nActual value: <element(s) not found> \nCall log:\nLocatorAssertions.to_have_text with timeout 5000ms\n  - waiting for locator(\"//aside//button[@data-testid='add-to-cart-cta-button']\")\n", "traceback": true}, "extra": {}, "file": {"name": "claim.py", "path": "/home/runner/work/Epic-Free-Games/Epic-Free-Games/epic/src/claim.py"}, "function": "run", "level": {"icon": "‚ùå", "name": "ERROR", "no": 40}, "line": 158, "message": "An error has been caught in function 'run', process 'MainProcess' (3641), thread 'MainThread' (140225620368256):", "module": "claim", "name": "__main__", "process": {"id": 3641, "name": "MainProcess"}, "thread": {"id": 140225620368256, "name": "MainThread"}, "time": {"repr": "2024-10-19 06:30:54.046618+08:00", "timestamp": 1729290654.046618}}}
